### PR TITLE
🍰 2019/kw22/refactor_backend_structure

### PR DESCRIPTION
**Pull request by Tirokk**
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-06-03T11:36:39Z" title="Monday, June 3rd 2019, 1:36:39 pm +02:00">Jun 3, 2019</time>_
_Merged <time datetime="2019-06-03T22:48:23Z" title="Tuesday, June 4th 2019, 12:48:23 am +02:00">Jun 4, 2019</time>_
---

### 2019/kw22/refactor_backend_structure
- refactored backend structure
- use and reference global config

### Issues

- relates https://github.com/Human-Connection/Human-Connection/issues/692

### Todo

- [ ] Middleware refactoring - not done in this PR

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -11,16 +11,16 @@
     "lint": "eslint src --config .eslintrc.js",
     "test": "run-s test:jest test:cucumber",
     "test:before:server": "cross-env GRAPHQL_URI=http://localhost:4123 GRAPHQL_PORT=4123 yarn run dev 2> /dev/null",
-    "test:before:seeder": "cross-env GRAPHQL_URI=http://localhost:4001 GRAPHQL_PORT=4001 DISABLED_MIDDLEWARES=permissions,activityPub yarn run dev 2> /dev/null",
+    "test:before:seeder": "cross-env GRAPHQL_URI=http://localhost:4001 GRAPHQL_PORT=4001 DEBUG=true DISABLED_MIDDLEWARES=permissions,activityPub yarn run dev 2> /dev/null",
```
Comment (Tirokk) [inline backend/package.json:14]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-03T21:06:40Z" title="Monday, June 3rd 2019, 11:06:40 pm +02:00">Jun 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -11,7 +11,7 @@
     "lint": "eslint src --config .eslintrc.js",
     "test": "run-s test:jest test:cucumber",
     "test:before:server": "cross-env GRAPHQL_URI=http://localhost:4123 GRAPHQL_PORT=4123 yarn run dev 2> /dev/null",
-    "test:before:seeder": "cross-env GRAPHQL_URI=http://localhost:4001 GRAPHQL_PORT=4001 DISABLED_MIDDLEWARES=permissions,activityPub yarn run dev 2> /dev/null",
+    "test:before:seeder": "cross-env GRAPHQL_URI=http://localhost:4001 GRAPHQL_PORT=4001 DEBUG=true DISABLED_MIDDLEWARES=permissions,activityPub yarn run dev 2> /dev/null",
```


Remove `DEBUG`. Maybe I'm missing sth, but I don't see the reason for it.

Comment (Tirokk) [reply backend/package.json:14]: > [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-06-03T21:22:18Z" title="Monday, June 3rd 2019, 11:22:18 pm +02:00">Jun 3, 2019</time>_

---


i enforced this in oder to remove the whole NODE_ENV check. Note the check is for DEBUG and DEBUG can only be set in a non production environment
`DEBUG: process.env.NODE_ENV !== 'production' && process.env.DEBUG === 'true'`

**Comment by Tirokk** [comment]
> [codecov[bot]](https://github.com/apps/codecov) **Authored by [codecov[bot]](https://github.com/apps/codecov)**
_<time datetime="2019-06-03T22:47:45Z" title="Tuesday, June 4th 2019, 12:47:45 am +02:00">Jun 4, 2019</time>_

---


# [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/724?src=pr&el=h1) Report
> Merging [#724](https://codecov.io/gh/Human-Connection/Human-Connection/pull/724?src=pr&el=desc) into [master](https://codecov.io/gh/Human-Connection/Human-Connection/commit/de6e2ee1f93c38e281d7e88fc8a3f8819b7a3a70?src=pr&el=desc) will **increase** coverage by `0.22%`.
> The diff coverage is `31.58%`.

[Impacted file tree graph](https://codecov.io/gh/Human-Connection/Human-Connection/pull/724?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #724      +/-   ##
==========================================
+ Coverage   22.95%   23.17%   +0.22%     
==========================================
  Files         163      164       +1     
  Lines        2309     2300       -9     
  Branches      119      119              
==========================================
+ Hits          530      533       +3     
+ Misses       1744     1732      -12     
  Partials       35       35
```


| [Impacted Files](https://codecov.io/gh/Human-Connection/Human-Connection/pull/724?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [backend/src/middleware/activityPubMiddleware.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/724/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvbWlkZGxld2FyZS9hY3Rpdml0eVB1Yk1pZGRsZXdhcmUuanM=) | `0% <ø> (ø)` | :arrow_up: |
| [backend/src/schema/resolvers/moderation.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/724/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvc2NoZW1hL3Jlc29sdmVycy9tb2RlcmF0aW9uLmpz) | `0% <ø> (ø)` | |
| [backend/src/schema/resolvers/comments.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/724/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvc2NoZW1hL3Jlc29sdmVycy9jb21tZW50cy5qcw==) | `0% <ø> (ø)` | |
| [backend/src/schema/types/index.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/724/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvc2NoZW1hL3R5cGVzL2luZGV4Lmpz) | `0% <ø> (ø)` | |
| [backend/src/schema/resolvers/socialMedia.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/724/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvc2NoZW1hL3Jlc29sdmVycy9zb2NpYWxNZWRpYS5qcw==) | `0% <ø> (ø)` | |
| [backend/src/schema/resolvers/shout.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/724/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvc2NoZW1hL3Jlc29sdmVycy9zaG91dC5qcw==) | `0% <ø> (ø)` | |
| [backend/src/middleware/userMiddleware.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/724/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvbWlkZGxld2FyZS91c2VyTWlkZGxld2FyZS5qcw==) | `0% <ø> (ø)` | :arrow_up: |
| [backend/src/schema/resolvers/notifications.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/724/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvc2NoZW1hL3Jlc29sdmVycy9ub3RpZmljYXRpb25zLmpz) | `0% <ø> (ø)` | |
| [backend/src/bootstrap/neo4j.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/724/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvYm9vdHN0cmFwL25lbzRqLmpz) | `100% <ø> (ø)` | :arrow_up: |
| [backend/src/schema/resolvers/user\_management.js](https://codecov.io/gh/Human-Connection/Human-Connection/pull/724/diff?src=pr&el=tree#diff-YmFja2VuZC9zcmMvc2NoZW1hL3Jlc29sdmVycy91c2VyX21hbmFnZW1lbnQuanM=) | `0% <ø> (ø)` | |
| ... and [20 more](https://codecov.io/gh/Human-Connection/Human-Connection/pull/724/diff?src=pr&el=tree-more) | |

**Review by Mogge — APPROVED** [review]
Sure! Refactoring all the configs. :+1: that's fine for me!

it's fine

**Pull request closed without merging**